src/Core/Components/DataGrid/FluentDataGrid.razor.cs

```diff
@@ -403,7 +408,7 @@ protected override Task OnParametersSetAsync()
             Pagination?.ItemsPerPage != _lastRefreshedPaginationState?.ItemsPerPage
             || Pagination?.CurrentPageIndex != _lastRefreshedPaginationState?.CurrentPageIndex;
 
-        var mustRefreshData = dataSourceHasChanged || paginationStateHasChanged;
+        var mustRefreshData = dataSourceHasChanged || paginationStateHasChanged || Loading is null;
```

Comment:
I'm not sure about this, but wanted your input. If Loading is set back to null, that would seem to imply that the user wants to re-load the data.